corpus_analysis/document.py

```diff
@@ -637,3 +640,83 @@ def update_metadata(self, new_metadata):
                         f" not '{new_metadata['date']}'"
                     ) from err
             setattr(self, key, new_metadata[key])
+
+    def get_char_list(self, cutoff_num=10):
+        """
+        given a document object, find a list of characters with their frequency in the novels
+        param: cutoff_num defaults to 10 for the thredshold for cutoffs based on named frequency
+        return: a list of tuples with character names in descending sorted order that occurs
+        more than the cutoff_num times in the document
+        >>> from corpus_analysis import document
+        >>> from pathlib import Path
+        >>> from gender_analysis import common
+        >>> document_metadata = {'author': 'Austen, Jane', 'title': 'Persuasion', 'date': '1818',
+        ...                      'filename': 'austen_persuasion.txt',
+        ...                      'filepath': Path(common.TEST_DATA_PATH, 'sample_novels',
+        ...                                       'texts', 'austen_persuasion.txt')}
+        >>> persuasion = document.Document(document_metadata)
+        >>> persuasion_chars = persuasion.get_char_list(20)
+        >>> persuasion_chars
+        [('Anne', 425), ('Captain Wentworth', 119), ('Lady Russell', 116), ('Charles', 115), ('Mary', 113), ('Sir Walter', 95), ('Elizabeth', 82), ('Elliot', 76), ('Louisa', 66), ('Henrietta', 65), ('Mrs Musgrove', 40), ('Mrs Smith', 36), ('Mrs Clay', 33), ('Miss Elliot', 32), ('Captain Benwick', 32), ('Wentworth', 31), ('Charles Hayter', 29), ('Mrs Croft', 27), ('Benwick', 26), ('Musgrove', 24), ('Uppercross', 23), ('Lady Dalrymple', 22), ('Captain Harville', 21)]
+        >>> len(persuasion_chars)
+        23
+        """
+
+        labels_char = []
+        labels = 'FACILITY,GPE,GSP,LOCATION,ORGANIZATION,PERSON'
+        document = self._load_document_text()
+        sentences = nltk.sent_tokenize(document)
+        for sent in sentences:
+            for chunk in nltk.ne_chunk(nltk.pos_tag(nltk.word_tokenize(sent))):
+                if hasattr(chunk, 'label'):
+                    labels_char.append((chunk.label(), ' '.join(c[0] for c in chunk)))
+        char_dict = {lab: {} for lab in labels.split(',')}
+        for ch in labels_char:
+            cat = char_dict[ch[0]]
+            cat[ch[1]] = cat.get(ch[1], 0) + 1
```

Comment:
Per my comment above, this is challenging to parse at first glance, and may present some difficulties to subsequent developers. I think a bit of explicit naming might help. Something like:
```
for character in labels_char:
            label, name = character
            char_dict[label][name] = char_dict[label].get(name, 0) + 1
```